src/super_gradients/training/utils/quantization_utils.py

```diff
@@ -0,0 +1,329 @@
+        num_calib_batches: int, number of batches to collect the statistics from (default=2).
+        percentile: float, percentile value to use when SgModel,quant_modules_calib_method='percentile'. Discarded when other methods are used (Default=99.99).
+
+        TODO: MAKE DDP SAFE
+    """
+
+    def __init__(self, calib_data_loader: DataLoader = None, num_calib_batches: int = 2, percentile: float = 99.99):
+        super().__init__(phase=Phase.PRE_TRAINING)
+        self.calib_data_loader = calib_data_loader
+        self.num_calib_batches = num_calib_batches
+        self.percentile = percentile
+
+    def __call__(self, context: PhaseContext):
+        self.calib_data_loader = self.calib_data_loader or context.valid_loader
+        calibrate_model(model=context.net,
+                        calib_data_loader=self.calib_data_loader,
+                        method=context.quant_modules_calib_method,
+                        num_calib_batches=self.num_calib_batches,
+                        percentile=self.percentile)
+
+        method_desc = context.quant_modules_calib_method + '_' + str(
+            self.percentile) if context.quant_modules_calib_method == 'percentile' else context.quant_modules_calib_method
+        if not context.ddp_silent_mode:
+            context.sg_logger.add_checkpoint(tag='ckpt_calibrated_' + method_desc + '.pth', state_dict={"net": context.net})
+
+
+class PostQATConversionCallback(PhaseCallback):
+    """
+    Post QAT training callback that saves the best checkpoint (i.e ckpt_best.pth) in onnx and trt engine formats.
+
+    Attributes:
+        dummy_input_size: (tuple) dummy input size for the ONNX conversion.
+    """
+    def __init__(self, dummy_input_size):
+        super().__init__(phase=Phase.POST_TRAINING)
+        self.dummy_input_size = dummy_input_size
+
+    def __call__(self, context: PhaseContext):
+        local_rank = get_local_rank()
+        with wait_for_the_master(local_rank):
+            if local_rank == 0:
+                best_ckpt_path = os.path.join(context.checkpoints_dir_path, "ckpt_best.pth")
+                onnx_path = os.path.join(context.checkpoints_dir_path, "ckpt_best.onnx")
+                trt_path = os.path.join(context.checkpoints_dir_path, "ckpt_best.engine")
+
+                load_checkpoint_to_model(ckpt_local_path=best_ckpt_path,
+                                         net=context.net,
+                                         load_weights_only=True,
+                                         load_ema_as_net=context.training_params.ema,
+                                         strict=True,
+                                         load_backbone=False
+                                         )
+
+                export_qat_onnx(context.net, onnx_path, self.dummy_input_size, context.per_channel_quant_modules)
+                save_trt_engine_from_onnx_ckpt(onnx_path, trt_path)
```

Comment:
why do we need this? 
SG is not supposed to take the checkpoint all the way to TRT. convert to ONNX, and that's it

Reply:
Removed.